src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/NativeRuntimeEventSource.cs

```diff
@@ -28,6 +28,11 @@ internal sealed partial class NativeRuntimeEventSource : EventSource
         [NonEvent]
         internal unsafe void ProcessEvent(uint eventID, uint osThreadID, DateTime timeStamp, Guid activityId, Guid childActivityId, ReadOnlySpan<byte> payload)
         {
+            if (!IsSupported)
```

Comment:
(nit) since this is an instance method, could this use `IsEnabled()` instead?

Reply:
I actually don't know - the runtime event source is weird in many ways, so using `IsSupported` felt like the safe option here.

Reply:
@noahfalk may advise better here. using IsEnabled looks the right way to check but I'll let @noahfalk comment on that.

Reply:
I think both IsSupported or IsEnabled() would work fine, but in the interest of simplicity (and performance) I'd say leave it as IsSupported. I'd suggest adding a comment to pull this out when its time to do #43657. This change feels like the quick and dirty version, #43657 would be the good version that prunes far more code and eliminates the need for this change.